src/main/kotlin/com/hibob/academy/filters/AuthenticationFilter.kt

```diff
@@ -0,0 +1,44 @@
+package com.hibob.academy.filters
+import io.jsonwebtoken.Claims
+import io.jsonwebtoken.Jws
+import io.jsonwebtoken.Jwts
+import jakarta.ws.rs.container.ContainerRequestContext
+import jakarta.ws.rs.container.ContainerRequestFilter
+import org.springframework.stereotype.Component
+import com.hibob.academy.service.SessionService.Companion.SECRET_KEY
+import jakarta.ws.rs.core.Response
+import jakarta.ws.rs.ext.Provider
+
+
+@Component
+@Provider
+class AuthenticationFilter : ContainerRequestFilter {
+    companion object {
+        private const val LOGIN_PATH = "jwt/users/login"
+        const val COOKIE_NAME = "ron_cookie_name"  // Replace with actual cookie name
+    }
+
+    override fun filter(requestContext: ContainerRequestContext) {
+        if (requestContext.uriInfo.path == LOGIN_PATH) return
+
+        // Verify the JWT token from the cookie
+        val cookie = requestContext.cookies[COOKIE_NAME]?.value
+
+        try {
+            val jwtClaims = verify(cookie) ?: throw Exception("Invalid or expired token")  // If claims are null, throw an exception
+        } catch (e: Exception) {
+            // Catch any exceptions (including null or verification failure) and abort with UNAUTHORIZED status
+            requestContext.abortWith(Response.status(Response.Status.UNAUTHORIZED).entity("Invalid or expired token").build())
+        }
```

Comment:
you already have an exception being thrown in the verify method? no need to throw a new exception here. just handle the null case in the verify method

Reply:
Got it 
Thanks Eli!